compiler/rustc_target/src/asm/powerpc.rs

```diff
@@ -58,6 +59,10 @@ impl PowerPCInlineAsmRegClass {
                 altivec: VecI8(16), VecI16(8), VecI32(4), VecF32(4);
                 vsx: F32, F64, VecI64(2), VecF64(2);
             },
+            // VSX is a superset of altivec.
+            Self::vsreg => types! {
+                vsx: F32, F64, VecI8(16), VecI16(8), VecI32(4), VecI64(2), VecF32(4), VecF64(2);
```

Comment:
technically this should also include 128-bit integer/fp types and `f16` since there are instructions for operating on those, but we can leave those out for now.